src/server/pfs/server/driver.go

```diff
@@ -280,7 +280,22 @@ func (d *driver) listRepo(ctx context.Context, includeAuth bool, repoType string
 		}
 		repoInfo.SizeBytesUpperBound = size
 
-		// TODO CORE-1111 check whether user has PROJECT_LIST_REPO on project or REPO_READ on repo.
+		// Check if the user can list repo within the project, otherwise check if they can read the repo.
+		if err := d.env.AuthServer.CheckProjectIsAuthorized(ctx, repoInfo.Repo.Project, auth.Permission_PROJECT_LIST_REPO); err != nil {
```

Comment:
I think it's worth optimizing this, because as it is we may be running the same auth queries for every repo in a project. One way to make this more performant is to first collect the user's auth info for each project and check that in processFunc, before running the individual REPO_READ query